zebrad/tests/acceptance.rs

```diff
@@ -243,10 +253,90 @@ fn start_args() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn persistent_mode() -> Result<()> {
+    zebra_test::init();
+    let (tempdir, _guard) = tempdir(ConfigMode::Persistent)?;
+
+    let mut child = get_child(&["-v", "start"], &tempdir)?;
+
+    // Run the program and kill it at 1 second
+    std::thread::sleep(Duration::from_secs(1));
+    child.kill()?;
+    let output = child.wait_with_output()?;
+
+    // Make sure the command was killed
+    assert!(output.was_killed());
+
+    // Check that we have persistent sled database
+    let cache_dir = tempdir.join("state");
+    assert!(cache_dir.read_dir()?.count() > 0);
+
+    Ok(())
+}
+
+#[test]
+fn ephemeral_mode() -> Result<()> {
+    zebra_test::init();
+    let (tempdir, _guard) = tempdir(ConfigMode::Ephemeral)?;
+
+    // Any free argument is valid
+    let mut child = get_child(&["start", "argument"], &tempdir)?;
+    // Run the program and kill it at 1 second
+    std::thread::sleep(Duration::from_secs(1));
+    child.kill()?;
+    let output = child.wait_with_output()?;
+
+    // Make sure the command was killed
+    assert!(output.was_killed());
+
+    let cache_dir = tempdir.join("state");
+    assert!(!cache_dir.exists() || cache_dir.read_dir()?.count() > 0);
+
+    Ok(())
+}
+
+#[test]
+fn misconfigured_ephemeral_mode() -> Result<()> {
+    zebra_test::init();
+
+    let dir = TempDir::new("zebrad_tests")?;
+    let cache_dir = dir.path().join("state");
+
+    // Write a configuration that has both cache_dir and ephemeral options set
+    let mut config = ZebradConfig::default();
+    config.state.ephemeral = true;
+    // Although cache_dir has a default value, we set it a new temp directory
+    // to test that it is empty later.
+    fs::create_dir(&cache_dir)?;
+    config.state.cache_dir = cache_dir.clone();
+    config.state.memory_cache_bytes = 256000000;
+    config.network.listen_addr = "127.0.0.1:0".parse()?;
```

Comment:
We're duplicating the test config defaults here.

That's ok for now, but I've opened #967 to remind us to clean it up in a future PR.